### PR TITLE
SAK-40963 Fixed overflow on GB item modal

### PR DESCRIPTION
**Pull request by adrianfish**

### CODE REVIEW

**Comment by adrianfish** [comment]
We have use !important on the height, otherwise Wicket just applies an inline height style and breaks the modal.

**Comment by fostersdesign** [comment]
@plukasew and I looked into this. By design, Wicket modals display internal scrollbars when the calculated modal height exceeds the window height on initial load (in the case of short viewport windows or long modal windows). 

SAK-34049 added the "overflow: initial !important;" on the "div.wicket-modal div.w_content_container" selector, which undoes this native behaviour of Wicket modals. 

Removing the overflow override should return this modal to the original behaviour (providing scrollbars when needed).

**Comment by adrianfish** [comment]
Isn't that sub optimal, though? Having scrollbars inside a div when you can just scroll the entire window, including the modal? What's the approved approach a11y wise?

**Comment by plukasew** [comment]
The modal is designed to be centred in the window both vertically and horizontally. As such, if the content is too large, it will scroll internally so the modal itself can stay entirely on screen. Presumably this is modelled after a traditional modal that would block all interaction with the rest of the application.

For the gradebook, most users will not encounter the inner scrolling as you need a pretty small window (unlikely for any serious use of the instructor view of the gradebook) or a very large modal to trigger it. From a maintenance standpoint, I would argue that we are best to leave it alone and let Wicket follow its native behaviour for these edge cases.

If we do decide it is best to override this behaviour, it can be done with javascript without the need for !important in css, and we could leave it as a configurable option to the developer so they can choose the best behaviour for each modal. This could also be done for just one axis or both (having to scroll the page horizontally to see a wide modal might not be as easy if you don't have a horizontal scroll wheel on your mouse).

From an accessibility standpoint, it could be argued that using a modal at all is problematic. I would agree with this, and would rather see an inline slideout panel appear instead of a modal popup, so that all the content on the page is visible and accessible. This is however a bit more work.

**Comment by adrianfish** [comment]
I just commented out that forced overflow and the auto height, and the scrolling turns back on, as you said. But, it's so clunky. The edit modal overflows my macbook pro so I have to scroll. One thing's for sure - that forced initial overflow broke Safari, before work on this ticket. You just can't access the buttons on the modal as scroll events pass through to the underlying content. Commenting those two lines out fixes Safari, but at the cost in my mind of visual unpleasantness. I agree that a slide out panel of content would be better, that's why I tried to lock the modal against the background so you could scroll the whole thing. So, maybe we just take the hit of clunky scrolling modals and talk about implementing slide outs for 20?

File: gradebookng/tool/src/webapp/styles/gradebook-grades.css [hunk 1]
```diff
@@ -705,12 +705,9 @@ div.wicket-modal div.w_right > div {
 div.wicket-modal {
   z-index: 999 !important;
   max-width: 100%;
+  position: absolute;
```
Comment (fostersdesign) [inline gradebookng/tool/src/webapp/styles/gradebook-grades.css]: Is this absolute positioning needed? I see that Wicket applies this to the modal inline already.

Comment (adrianfish) [reply gradebookng/tool/src/webapp/styles/gradebook-grades.css]: Yes, it's needed to make Safari work properly. Without it, you get this:
screen shot 2019-01-16 at 09 33 28
Hidden content and no scrolling of the modal into view.

With it you get this:
screen shot 2019-01-16 at 09 36 29
You can scroll the modal into view and see the buttons.

Comment (fostersdesign) [reply gradebookng/tool/src/webapp/styles/gradebook-grades.css]: Okay. Thanks for confirming. 

What you have now looks good to me then.

Comment (adrianfish) [reply gradebookng/tool/src/webapp/styles/gradebook-grades.css]: @fostersdesign Could you review, please? I can merge it then.

Comment (fostersdesign) [reply gradebookng/tool/src/webapp/styles/gradebook-grades.css]: I have already changed my review to "Approved". I just don't have any authority here, so it doesn't count as a required review.

Maybe ask @ern or @steveswinsburg for their approval too?

Comment (adrianfish) [reply gradebookng/tool/src/webapp/styles/gradebook-grades.css]: @ern Any chance?

**Pull request merged**